File: 47 Hi-Lo/python/hilo.py
#!/usr/bin/env python3

QUESTION_PROMPT='? '

def play():
    print('HI LO')
    print('CREATIVE COMPUTING  MORRISTOWN, NEW JERSEY\n\n\n')
    print('THIS IS THE GAME OF HI LO.\n')
    print('YOU WILL HAVE 6 TRIES TO GUESS THE AMOUNT OF MONEY IN THE')
    print('HI LO JACKPOT, WHICH IS BETWEEN 1 AND 100 DOLLARS.  IF YOU')
    print('GUESS THE AMOUNT, YOU WIN ALL THE MONEY IN THE JACKPOT!')
    print('THEN YOU GET ANOTHER CHANCE TO WIN MORE MONEY.  HOWEVER,')
    print('IF YOU DO NOT GUESS THE AMOUNT, THE GAME ENDS.\n\n')

    total_winnings = 0
    while True:
        print('PLAY AGAIN (YES OR NO)', end=QUESTION_PROMPT)
        answer = input().upper()
        if answer != 'YES':
            break

    print('\nSO LONG.  HOPE YOU ENJOYED YOURSELF!!!')

if __name__ == '__main__':
    play()

File: 00_Alternate_Languages/47_Hi-Lo/ruby/hi_lo.rb
#!/usr/bin/env ruby
MAX_TRIES = 6
RANGE = (1..100)

def intro
  puts <<~END_OF_INTRO
    #{'HI LO'.center(74)}
    #{"CREATIVE COMPUTING  MORRISTOWN, NEW JERSEY\n\n\n".center(76)}
    THIS IS THE GAME OF HI LO.\n
    YOU WILL HAVE #{MAX_TRIES} TRIES TO GUESS THE AMOUNT OF MONEY IN THE
    HI LO JACKPOT, WHICH IS BETWEEN #{RANGE.min} AND #{RANGE.max} DOLLARS.  IF YOU
    GUESS THE AMOUNT, YOU WIN ALL THE MONEY IN THE JACKPOT!
    THEN YOU GET ANOTHER CHANCE TO WIN MORE MONEY.  HOWEVER,
    IF YOU DO NOT GUESS THE AMOUNT, THE GAME ENDS.\n\n
  END_OF_INTRO
end

def make_guess
  puts 'YOUR GUESS?'
  @guess = gets.to_i
end

def check_guess
  if @guess == @number
    @guessed_correctly = true
    @total_winnings += @number
    puts <<~END_OF_WIN_TEXT
      GOT IT!!!!!!!!!!   YOU WIN #{@number} DOLLARS.
      YOUR TOTAL WINNINGS ARE NOW #{@total_winnings} DOLLARS.
    END_OF_WIN_TEXT
  else
    puts "YOUR GUESS IS TOO #{@guess > @number ? 'HIGH' : 'LOW'}.\n\n"
  end
end

def blew_it
  @total_winnings = 0
  puts "YOU BLEW IT...TOO BAD...THE NUMBER WAS #{@number}"
end

def outro
  puts "\nSO LONG.  HOPE YOU ENJOYED YOURSELF!!!"
end

intro
@total_winnings = 0
loop do
  @guessed_correctly = false
  @number = rand(RANGE)
  MAX_TRIES.times do
    make_guess
    check_guess
    break if @guessed_correctly
  end
  blew_it unless @guessed_correctly
  puts "\nPLAY AGAIN (YES OR NO)?"
  break if gets.start_with?(/n/i)
end
outro
